src/main/java/com/google/api/codegen/transformer/php/PhpGapicSurfaceTransformer.java

```diff
@@ -228,6 +255,117 @@ private ViewModel buildClientViewModel(GapicInterfaceContext context) {
     return result;
   }
 
+  private ViewModel buildDescriptorConfigViewModel(GapicInterfaceContext context) {
+    return DescriptorConfigView.newBuilder()
+        .pageStreamingDescriptors(pageStreamingTransformer.generateDescriptors(context))
+        .hasPageStreamingMethods(context.getInterfaceConfig().hasPageStreamingMethods())
+        .hasBatchingMethods(context.getInterfaceConfig().hasBatchingMethods())
+        .longRunningDescriptors(createLongRunningDescriptors(context))
+        .hasLongRunningOperations(context.getInterfaceConfig().hasLongRunningOperations())
+        .grpcStreamingDescriptors(createGrpcStreamingDescriptors(context))
+        .interfaceKey(context.getInterface().getFullName())
+        .templateFileName(DESCRIPTOR_CONFIG_TEMPLATE_FILENAME)
+        .outputPath(generateConfigOutputPath(context, "descriptor_config"))
+        .build();
+  }
+
+  private ViewModel buildRestConfigViewModel(GapicInterfaceContext context) {
+    return RestConfigView.newBuilder()
+        .templateFileName(REST_CONFIG_TEMPLATE_FILENAME)
+        .outputPath(generateConfigOutputPath(context, "rest_client_config"))
+        .interfaceKey(context.getInterface().getFullName())
+        .apiMethods(generateRestMethodConfigViews(context))
+        .build();
+  }
+
+  private List<RestMethodConfigView> generateRestMethodConfigViews(GapicInterfaceContext context) {
+    List<RestMethodConfigView> configViews = new ArrayList<>();
+
+    for (MethodModel methodModel : context.getSupportedMethods()) {
+      GapicMethodContext methodContext = context.asDynamicMethodContext(methodModel);
+
+      if (!methodContext.getMethodConfig().isGrpcStreaming()) {
+        configViews.add(generateRestMethodConfigView(methodContext));
+      }
+    }
+
+    return configViews;
+  }
+
+  private RestMethodConfigView generateRestMethodConfigView(GapicMethodContext context) {
+    RestMethodConfigView.Builder restMethodConfig = RestMethodConfigView.newBuilder();
+    Method method = context.getMethod();
+    Map<String, String> httpMethodMap = new HashMap<>();
+    HttpRule httpRule = getHttpRule(method.getOptionFields());
+    String body = httpRule.getBody();
+
+    httpMethodMap.put("get", httpRule.getGet());
+    httpMethodMap.put("put", httpRule.getPut());
+    httpMethodMap.put("delete", httpRule.getDelete());
+    httpMethodMap.put("post", httpRule.getPost());
+    httpMethodMap.put("patch", httpRule.getPatch());
+
+    for (Map.Entry<String, String> entry : httpMethodMap.entrySet()) {
```

Comment:
This feels like an odd way to use a map - we don't do any lookups, just use it as a way to create a 2-tuple. I am not sure what the correct restructuring is - maybe this whole for loop should be a helper function?

Reply:
We discussed this item offline and decided the current strategy is acceptable.